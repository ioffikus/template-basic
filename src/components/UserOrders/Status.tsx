import React from 'react';
import { Popover } from 'antd';
import TransactionInfo from './TransactionInfo';

interface IProps {
  status: string;
  transactionDetails: string;
  transactionStatus: string;
  ui: Alicanto.Models.UI;
  getRootRef: () => HTMLElement;
}

interface IState {}

class Status extends React.Component<IProps, IState> {
  render() {
    const { status, transactionDetails, transactionStatus, getRootRef, ui } = this.props;
    const isTransaction = transactionDetails || transactionStatus;
    return (
      <div className="root">
        <style jsx>{`
          .root :global(.icon) {
            width: 17px;
            height: 17px;
            text-align: center;
            margin: 0 10px;
            display: inline-block;
            font-size: 0.8em;
            font-weight: bold;
            background-color: #e5ecf4;
            border-radius: 50%;
            cursor: pointer;
          }
        `}</style>
        {`Status: ${status}`}
        {isTransaction && (
          <Popover
            overlayClassName="status-popover"
            getPopupContainer={getRootRef}
            content={<TransactionInfo ui={ui} details={transactionDetails} status={transactionStatus} />}
          >
            <span className="icon">i</span>
          </Popover>
        )}
      </div>
    );
  }
}

export default Status;
