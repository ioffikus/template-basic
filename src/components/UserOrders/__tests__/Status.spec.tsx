import React from 'react';
import { shallow } from 'enzyme';
import uiStateMock from 'src/store/__mocks__/uiStore';
import Status from '../Status';

describe(null, () => {
  it('render() correct', () => {
    const status = 'dymmy status';
    const transactionDetails = 'dymmy transactionDetails';
    const transactionStatus = 'dymmy transactionStatus';
    const getRootRef = jest.fn();
    const wrapper = shallow(
      <Status
        status={status}
        transactionDetails={transactionDetails}
        transactionStatus={transactionStatus}
        getRootRef={getRootRef}
        ui={uiStateMock}
      />,
    );
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('render() without status', () => {
    const status = 'dymmy status';
    const transactionDetails = '';
    const transactionStatus = '';
    const getRootRef = jest.fn();
    const wrapper = shallow(
      <Status
        status={status}
        transactionDetails={transactionDetails}
        transactionStatus={transactionStatus}
        getRootRef={getRootRef}
        ui={uiStateMock}
      />,
    );
    expect(wrapper.render()).toMatchSnapshot();
  });
});
