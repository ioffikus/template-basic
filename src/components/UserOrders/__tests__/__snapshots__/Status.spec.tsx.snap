// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render() correct 1`] = `
<div
  class="jsx-1044737517 root"
>
  Status: dymmy status
  <span
    class="jsx-1044737517 icon"
  >
    i
  </span>
</div>
`;

exports[`render() without status 1`] = `
<div
  class="jsx-1044737517 root"
>
  Status: dymmy status
</div>
`;
